pyupgrade/_plugins/typing_pep585.py

```diff
@@ -36,7 +36,7 @@ def visit_Attribute(
     if (
             _should_rewrite(state) and
             isinstance(node.value, ast.Name) and
-            node.value.id == 'typing' and
+            state.as_imports.get(node.value.id) == 'typing' and
```

Comment:
Note that this breaks cases like we see in `typing_pep604_test.py` below, where we use `typing` without first importing it. I think that's okay since that code is already broken in real-life, but if we wanted to avoid making those changes we could trivially change this to:

```suggestion
        state.as_imports.get(node.value.id, node.value.id) == 'typing'
```